File: go/account.go
package nexcore

import (
	"encoding/json"
)

// AccountNamespace implements the v1 账户 read endpoints.
//
// 对应 NexCore 账户模块的公开只读接口:
//
//	GET /api/v1/account/balance          GetBalance        查询账户余额
//	GET /api/v1/account/deposit-address  GetDepositAddress 查询/获取充值地址
//
// 鉴权:X-API-Key + X-Secret-Key 双 header(用 cfg.APIKey / cfg.APISecret).
// account 与 vcard 命名空间共用同一对 MPK 商户密钥.
type AccountNamespace struct {
	c *Client
}

// headers 构造 X-API-Key + X-Secret-Key(复用与 energy 一致的双密钥读模式).
func (n *AccountNamespace) headers() (map[string]string, error) {
	if n.c.cfg.APIKey == "" || n.c.cfg.APISecret == "" {
		return nil, &Error{Message: "APIKey / APISecret not configured", Code: -1}
	}
	return map[string]string{
		"X-API-Key":    n.c.cfg.APIKey,
		"X-Secret-Key": n.c.cfg.APISecret,
	}, nil
}

// GetBalance returns the account balance.
//
// GET /api/v1/account/balance
func (n *AccountNamespace) GetBalance() (json.RawMessage, error) {
	h, err := n.headers()
	if err != nil {
		return nil, err
	}
	return n.c.transport.do("GET", "/api/v1/account/balance", &requestOpts{Headers: h})
}

// GetDepositAddress returns the account deposit address(es).
//
// GET /api/v1/account/deposit-address
func (n *AccountNamespace) GetDepositAddress() (json.RawMessage, error) {
	h, err := n.headers()
	if err != nil {
		return nil, err
	}
	return n.c.transport.do("GET", "/api/v1/account/deposit-address", &requestOpts{Headers: h})
}

File: php/src/Namespaces/Account.php
<?php
/**
 * 账户命名空间.
 *
 * 对应平台「账户」模块的 v1 公开接口(余额 / 充值地址查询).
 *
 * 鉴权:X-API-Key + X-Secret-Key 双 header(MPK 商户密钥,与 VCard 命名空间共用).
 * 凭据来自控制台「账号 → API 密钥」,SDK 配置字段 api_key / api_secret.
 *
 *   GET /api/v1/account/balance          getBalance          查询账户余额
 *   GET /api/v1/account/deposit-address  getDepositAddress   查询充值地址
 */

declare(strict_types=1);

namespace NexCore\Namespaces;

use NexCore\Client;
use NexCore\NexCoreError;

class Account
{
    private Client $client;

    public function __construct(Client $c) { $this->client = $c; }

    /**
     * 组装双密钥鉴权头(与 Energy::authHeaders() 同形式,但读 api_key / api_secret).
     *
     * @return list<string>
     * @throws NexCoreError 凭据未配置
     */
    private function authHeaders(): array
    {
        $k = $this->client->get('api_key');
        $s = $this->client->get('api_secret');
        if (!$k || !$s) {
            throw new NexCoreError('api_key / api_secret not configured', -1);
        }
        return ["X-API-Key: $k", "X-Secret-Key: $s"];
    }

    /**
     * 查询账户余额.
     *
     * GET /api/v1/account/balance
     *
     * @return array<string, mixed> {balance, frozen, currency, ...}
     * @throws NexCoreError
     */
    public function getBalance(): array
    {
        return $this->client->http->request('GET', '/api/v1/account/balance', [
            'headers' => $this->authHeaders(),
        ]);
    }

    /**
     * 查询充值地址.
     *
     * GET /api/v1/account/deposit-address
     *
     * @return array<string, mixed> {address, chain, ...}
     * @throws NexCoreError
     */
    public function getDepositAddress(): array
    {
        return $this->client->http->request('GET', '/api/v1/account/deposit-address', [
            'headers' => $this->authHeaders(),
        ]);
    }
}
